cognite/neat/core/data_stores/metrics.py

```diff
@@ -0,0 +1,39 @@
+from typing import Iterable
+
+from cognite.client import CogniteClient
+from prometheus_client import REGISTRY, Counter, Gauge, Metric
+
+
+class NeatMetricsCollector:
+    def __init__(self, name: str, cdf_client: CogniteClient = None) -> None:
+        self.name = name
+        self.metrics: dict[str, Gauge | Counter] = {}
+
+    def register_metric(
+        self, metric_name: str, metric_description: str = "", m_type: str = "gauge", metric_labels: list[str] = None
+    ) -> None:
+        """Register metric in prometheus"""
+        metric_labels = [] if metric_labels is None else metric_labels
+        metric = None
+        metric_name = f"neat_workflow_{self.name}_{metric_name}"
+        if metric_name in REGISTRY._names_to_collectors:
+            self.metrics[metric_name] = REGISTRY._names_to_collectors[metric_name]
+        else:
+            if m_type == "gauge":
+                metric = Gauge(metric_name, metric_description, metric_labels)
+            elif m_type == "counter":
+                metric = Counter(metric_name, metric_description, metric_labels)
+            if metric:
+                self.metrics[metric_name] = metric
```

Comment:
I found this slightly hard to read, so just a 🏌 suggestion to make it more readable.
```suggestion
        metric_labels = [] if metric_labels is None else metric_labels

        metric_name = f"neat_workflow_{self.name}_{metric_name}"
        if metric_name in REGISTRY._names_to_collectors:
            self.metrics[metric_name] = REGISTRY._names_to_collectors[metric_name]
            return

        metric = None
        if m_type == "gauge":
            metric = Gauge(metric_name, metric_description, metric_labels)
        elif m_type == "counter":
            metric = Counter(metric_name, metric_description, metric_labels)

        if metric:
            self.metrics[metric_name] = metric
```